File: app/src/main/java/eu/indiewalkabout/mathbrainer/ui/HighscoresViewModelFactory.java
package eu.indiewalkabout.mathbrainer.ui;


import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

public class HighscoresViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    /*
    public HighscoresViewModelFactory(String foodlistType) {
        this.foodlistType = foodlistType;
    }
    */


    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new HighScoresViewModel();
    }

}

File: app/src/main/java/eu/indiewalkabout/mathbrainer/ui/HighscoresViewModelFactory.kt
package eu.indiewalkabout.mathbrainer.ui


import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class HighscoresViewModelFactory : ViewModelProvider.NewInstanceFactory() {

    /*
    public HighscoresViewModelFactory(String foodlistType) {
        this.foodlistType = foodlistType;
    }
    */


    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return HighScoresViewModel() as T
    }

}
